node/src/types/block/approvals_hashes.rs

```diff
@@ -167,3 +167,73 @@ pub(crate) enum ApprovalsHashesValidationError {
         value_in_proof: Digest,
     },
 }
+
+#[cfg(test)]
+mod specimen_support {
+    use crate::testing::specimen::{
+        largest_variant, vec_of_largest_specimen, vec_prop_specimen, LargestSpecimen, SizeEstimator,
+    };
+
+    use super::ApprovalsHashes;
+    use casper_execution_engine::storage::trie::{
+        merkle_proof::{TrieMerkleProof, TrieMerkleProofStep},
+        Pointer,
+    };
+    use casper_hashing::Digest;
+    use casper_types::{bytesrepr::Bytes, CLValue, Key, StoredValue};
+    use core::convert::TryInto;
+    use once_cell::sync::OnceCell;
+    use std::collections::BTreeMap;
+
+    impl LargestSpecimen for ApprovalsHashes {
+        fn largest_specimen<E: SizeEstimator>(estimator: &E) -> Self {
+            let data = {
+                let mut map = BTreeMap::new();
+                map.insert("key1", Digest::hash(""));
```

Comment:
These keys need to be from constants, specifically `APPROVALS_CHECKSUM_NAME` and `EXECUTION_RESULTS_CHECKSUM_NAME`. These are longer than the examples.

Reply:
Also consider using `LargestSpecimen::largest_specimen()` for the hash.